python/cudf/cudf/tests/test_udf_masked_ops.py

```diff
@@ -802,6 +802,18 @@ def func(x):
     run_masked_udf_series(func, data, check_dtype=False)
 
 
+@pytest.mark.parametrize(
+    "data", [[1, -1, 0, cudf.NA], [1, 0, -1, cudf.NA], [True, False, cudf.NA]]
```

Comment:
Please can you test "interesting" corner cases.

In particular:

1. Input series should be both signed and unsigned integer dtypes, and also at least one float type
2. Values should include `numpy.iinfo(dtype).min` and `numpy.iinfo(dtype).max` (for integer dtypes) and `ffinfo.min/max` along with `nan` and `inf` and `-inf` for floats.

Reply:
I added these test cases and indeed the overflow case where `abs(np.iinfo('int64').min)` won't fit in an `int64` fails. 

To get around this I've added code that upcasts to an unsigned type of the same bit width. Let me know if you think this is an appropriate solution.